src/app.service.ts

```diff
@@ -1,16 +1,34 @@
 import { Injectable } from '@nestjs/common';
 import { parseGameStateFromFile } from './common/parseConfigFile';
 import { join } from 'path';
-import { GameState } from './common/gameState';
+import {
+  GameState,
+  updateGameState,
+  initNewGameState,
+  DEFAULT_BOARD_SIZE,
+} from './common/gameState';
+import { Coordinates } from './common/utils';
 
 const configPathFromDistDir = '../gameStateFile.json';
 
 @Injectable()
 export class AppService {
-  getBoardStateFromFile(): { gameState: GameState } {
+  getBoardStateFromFile(): GameState {
     const gameState = parseGameStateFromFile(
       join(__dirname, configPathFromDistDir),
     );
-    return { gameState };
+    return gameState;
+  }
+
+  updateGameState(gameState: GameState, coordinates: Coordinates): GameState {
+    try {
+      updateGameState(gameState, coordinates);
```

Comment:
`updateGameState` is supposed to return a new game state, not to modify its parameters (which is considered a bad practice in functional programming, and is even forbiddent by some runtimes)

Reply:
Oh yes, i forgot to use the game state returned by updateGameState.

Is something like this okay for you ?
```
try {
  const updatedGameState = updateGameState(gameState, coordinates);
  return updatedGameState;
} catch {
  return gameState;
}
```